packages/zoe/test/unitTests/contracts/test-callSpread.js

```diff
@@ -0,0 +1,664 @@
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now => {
+          const price = priceFromSchedule(now);
+          return strikeAmountMath.make(price * underlyingValue);
+        });
+    },
+    priceAtTime: (timeStamp, underlyingAmount) => {
+      const { promise, resolve } = makePromiseKit();
+
+      underlyingAmountMath.getValue(underlyingAmount);
+      E(timer).setWakeup(
+        timeStamp,
+        harden({
+          wake: () => {
+            return resolve(priceAuthority.getCurrentPrice(underlyingAmount));
+          },
+        }),
+      );
+      return promise;
+    },
+  };
+  return priceAuthority;
+}
+
+// Underlying is in Simoleans. Collateral, strikePrice and Payout are in bucks.
+// Value is in Moola. The price oracle takes an amount in Underlying, and
+// gives the value in Moola.
+test('callSpread below Strike1', async t => {
+  const {
+    moolaIssuer,
+    simoleanIssuer,
+    moola,
+    simoleans,
+    bucksIssuer,
+    bucksMint,
+    bucks,
+    zoe,
+    amountMaths,
+  } = setup();
+  const installation = await installationPFromSource(zoe, callSpread);
+  const invitationIssuer = await E(zoe).getInvitationIssuer();
+
+  // Alice will create and fund a call spread contract, and give the invitations
+  // to Bob and Carol. Bob and Carol will promptly schedule collection of funds.
+  // The spread will then mature at a low price, and carol will get paid.
+
+  // Setup Alice
+  const aliceBucksPayment = bucksMint.mintPayment(bucks(300));
+  // Setup Bob
+  const bobBucksPurse = bucksIssuer.makeEmptyPurse();
+  // Setup Carol
+  const carolBucksPurse = bucksIssuer.makeEmptyPurse();
+
+  // Alice creates a callSpread instance
+  const issuerKeywordRecord = harden({
+    Underlying: simoleanIssuer,
+    Collateral: bucksIssuer,
+    Strike: moolaIssuer,
+    Options: invitationIssuer,
+  });
+
+  const manualTimer = buildManualTimer(console.log, 1);
+  const priceAuthority = makeFakePriceAuthority(
+    manualTimer,
+    amountMaths.get('simoleans'),
+    amountMaths.get('moola'),
+    [
+      { time: 0, price: 20 },
+      { time: 1, price: 35 },
+      { time: 2, price: 15 },
+      { time: 3, price: 28 },
+    ],
+  );
+  // underlying is 2 Simoleans, strike range is 30-50 (doubled)
+  const terms = harden({
+    expiration: 3,
+    underlyingAmount: simoleans(2),
+    priceAuthority,
+    strikePrice1: moola(60),
+    strikePrice2: moola(100),
+    settlementAmount: bucks(300),
+  });
+  const { creatorInvitation } = await zoe.startInstance(
+    installation,
+    issuerKeywordRecord,
+    terms,
+  );
+
+  const optionAmount = await invitationIssuer.getAmountOf(creatorInvitation);
+  const longOptionAmount = optionAmount.value[0].LongOption;
+  const shortOptionAmount = optionAmount.value[0].ShortOption;
+
+  const aliceProposal = harden({
+    want: { LongOption: longOptionAmount, ShortOption: shortOptionAmount },
+    give: { Collateral: bucks(300) },
```

Comment:
should we call this margin instead?

Reply:
This is a fully collateralized call spread, so there's no margin.

Reply:
I think people call "collateral" margin even when it is fully collateralized, but I could be wrong. Take a look at Joe's doc.